clib/src/main/resources/scala-native/signal.c

```diff
@@ -19,5 +19,3 @@ int scalanative_sigint() { return SIGINT; }
 int scalanative_sigsegv() { return SIGSEGV; }
 
 int scalanative_sigterm() { return SIGTERM; }
-
-int scalanative_sigusr1() { return SIGUSR1; }
```

Comment:
I have not run a program, but I think that deleting line 23 here
will cause lines 41 & 42 in signal.scala to have a
"Missing symbol" at link time.

I believe the intended change is to remove this line
when the deprecation grace period of lines 41/32 in signal.scala
expires and those lines are removed.

Reply:
The linkage between the `.c.o` and `.ll.o`(which contains the `extern`s) happens at link time so all the C object code is essentially a big blob with no namspace. This is why we can only have the C code in one place.

I would agree if we omitted `posixlib` for Windows compiling for example. I am hoping we can just remove the Scala interface in `clib` that is deprecated in the `0.4.2` release. I don't think a long deprecation cycle is warranted based on fact that we have very little usage. I also suspect or at least hope that we can have some minor releases in the mean time before we officially change things for Windows. If we do change sooner then the code would compile but fail to link which is like an error on deprecation but only on the Windows platform that shouldn't have the POSIX constant anyway.

You can move a C file from one project to another and see that it still works to satisfy your curiosity.

Reply:
Given the information in the reply just above and further study, I think the problem statement
for this point is:

```
 SIGUSR1 is a Gnu C extension. It is not  part of the 
`EEE Std 1003.1-2017 (Revision of IEEE Std 1003.1-2008)` specification, so it should
not be included in clib.  It is part of clib in SN 0.4.0 because of the impure initial
seeding of clib.

The presence of this declaration prevent compilation on Windows and possibly 
other non-GnuC systems, so it should be surgically removed after a short deprecation
cycle on Gnu C Linux.
```

Given that problem statement, or a close evolution of it, something similar to the
following would allow compilation on Windows (and non-GnuC systems) yet
not introduce a decidedly unfriendly link time "Missing Symbol"  error.
(By my understanding, "Deprecated" means "continues to work for a short time"
not a link time error.)

```
// not compiled, for concept
#ifdef __GNUC__
// Remove after deprecated & only caller in posix signal.scala is removed.
int scalanative_sigusr1() { return SIGUSR1; }
#endif
```

The technique is general to deprecating & removing other GnuC extensions in clib.

Reply:
This is a fine idea but that is why I moved it from `clib` to `posixlib`. This way windows will not see it as explained above. I think if we get to a point were this is not removed before Win support is added we can deal with it then. My guess is Win support is a ways off.

Reply:
OK, now I see! Duh!  This is basically a (well hidden) move from clib to posixlib.  Searching again,
I see SIGUSR1 in the 2018 Open Group Posix spec.

A good change overall.   

I got lost in the steps to the dance. When I saw the deprecation code in
clib signal.scala I expected, according to the current orginization, for the
implementation corresponding to `@name scalanative_sigusr1` to be
in clib signal.c.   

Yes, it works due to C's flat namespace and our conventions, but it
is hard to follow the elegant intervention.

Perhaps adding a comment to the clib/libc signal.scala. Someday,
someone will sing your praises.

```
@name("scalanative_sigusr1") // implemented in posix signal.c
```